App crashes when context passed to network utils is null

**Summary:** 

As per @nicolas-raoul's crash reports the app crashes when the context is null. 

**System logs:**

```
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'android.content.Context android.content.Context.getApplicationContext()' on a null object reference
at fr.free.nrw.commons.utils.NetworkUtils.isInternetConnectionEstablished(NetworkUtils.java:12)
at fr.free.nrw.commons.explore.images.SearchImageFragment.updateImageList(SearchImageFragment.java:133)
at fr.free.nrw.commons.explore.SearchActivity.lambda$setTabs$1$SearchActivity(SearchActivity.java:121)
at fr.free.nrw.commons.explore.SearchActivity$$Lambda$1.accept(Unknown Source)
at io.reactivex.internal.observers.LambdaObserver.onNext(LambdaObserver.java:63)
... 10 more
java.lang.NullPointerException: Attempt to invoke virtual method 'android.content.Context android.content.Context.getApplicationContext()' on a null object reference
at fr.free.nrw.commons.utils.NetworkUtils.isInternetConnectionEstablished(NetworkUtils.java:12)
at fr.free.nrw.commons.explore.images.SearchImageFragment.updateImageList(SearchImageFragment.java:133)
at fr.free.nrw.commons.explore.SearchActivity.lambda$setTabs$1$SearchActivity(SearchActivity.java:121)
at fr.free.nrw.commons.explore.SearchActivity$$Lambda$1.accept(Unknown Source)
at io.reactivex.internal.observers.LambdaObserver.onNext(LambdaObserver.java:63)
at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.drainNormal(ObservableObserveOn.java:200)
at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.run(ObservableObserveOn.java:252)
at io.reactivex.android.schedulers.HandlerScheduler$ScheduledRunnable.run(HandlerScheduler.java:119)
at android.os.Handler.handleCallback(Handler.java:751)
at android.os.Handler.dispatchMessage(Handler.java:95)
at android.os.Looper.loop(Looper.java:154)
at android.app.ActivityThread.main(ActivityThread.java:6186)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)
```

**Device and Android version:** 

7.1.2
 
**Commons app version:** 

2.9.0.118

**Would you like to work on the issue?**

Yes

--- a/app/src/main/java/fr/free/nrw/commons/utils/NetworkUtils.java
+++ b/app/src/main/java/fr/free/nrw/commons/utils/NetworkUtils.java
@@ -1,21 +1,34 @@
 package fr.free.nrw.commons.utils;
 
 
+import android.annotation.SuppressLint;
 import android.content.Context;
 import android.net.ConnectivityManager;
 import android.net.NetworkInfo;
 
+import javax.annotation.Nullable;
+
 public class NetworkUtils {
 
-    public static boolean isInternetConnectionEstablished(Context context) {
+    /**
+     * https://developer.android.com/training/monitoring-device-state/connectivity-monitoring#java
+     * Check if internet connection is established.
+     * @param context context passed to this method could be null.
+     * @return Returns current internet connection status. Returns false if null context was passed.
+     */
+    @SuppressLint("MissingPermission")
+    public static boolean isInternetConnectionEstablished(@Nullable Context context) {
+        if (context == null) {
+            return false;
+        }
+
         ConnectivityManager cm =
-                (ConnectivityManager)context.getApplicationContext().getSystemService(Context.CONNECTIVITY_SERVICE);
+                (ConnectivityManager) context.getApplicationContext().getSystemService(Context.CONNECTIVITY_SERVICE);
 
         if (cm == null) {
             return false;
         }
         NetworkInfo activeNetwork = cm.getActiveNetworkInfo();
-        return activeNetwork != null &&
-                activeNetwork.isConnectedOrConnecting();
+        return activeNetwork != null && activeNetwork.isConnectedOrConnecting();
     }
 }

--- a/app/src/test/kotlin/fr/free/nrw/commons/utils/NetworkUtilsTest.java
+++ b/app/src/test/kotlin/fr/free/nrw/commons/utils/NetworkUtilsTest.java
@@ -1,0 +1,71 @@
+package fr.free.nrw.commons.utils;
+
+import android.app.Application;
+import android.content.Context;
+import android.net.ConnectivityManager;
+import android.net.NetworkInfo;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class NetworkUtilsTest {
+
+    @Before
+    public void setUp() throws Exception {
+    }
+
+    @Test
+    public void testInternetConnectionEstablished() {
+        Context mockContext = mock(Context.class);
+        Application mockApplication = mock(Application.class);
+        ConnectivityManager mockConnectivityManager = mock(ConnectivityManager.class);
+        NetworkInfo mockNetworkInfo = mock(NetworkInfo.class);
+        when(mockNetworkInfo.isConnectedOrConnecting())
+                .thenReturn(true);
+        when(mockConnectivityManager.getActiveNetworkInfo())
+                .thenReturn(mockNetworkInfo);
+        when(mockApplication.getSystemService(Context.CONNECTIVITY_SERVICE))
+                .thenReturn(mockConnectivityManager);
+        when(mockContext.getApplicationContext()).thenReturn(mockApplication);
+        boolean internetConnectionEstablished = NetworkUtils.isInternetConnectionEstablished(mockContext);
+        assertTrue(internetConnectionEstablished);
+    }
+
+    @Test
+    public void testInternetConnectionNotEstablished() {
+        Context mockContext = mock(Context.class);
+        Application mockApplication = mock(Application.class);
+        ConnectivityManager mockConnectivityManager = mock(ConnectivityManager.class);
+        NetworkInfo mockNetworkInfo = mock(NetworkInfo.class);
+        when(mockNetworkInfo.isConnectedOrConnecting())
+                .thenReturn(false);
+        when(mockConnectivityManager.getActiveNetworkInfo())
+                .thenReturn(mockNetworkInfo);
+        when(mockApplication.getSystemService(Context.CONNECTIVITY_SERVICE))
+                .thenReturn(mockConnectivityManager);
+        when(mockContext.getApplicationContext()).thenReturn(mockApplication);
+        boolean internetConnectionEstablished = NetworkUtils.isInternetConnectionEstablished(mockContext);
+        assertFalse(internetConnectionEstablished);
+    }
+
+    @Test
+    public void testInternetConnectionStatusForNullContext() {
+        boolean internetConnectionEstablished = NetworkUtils.isInternetConnectionEstablished(null);
+        assertFalse(internetConnectionEstablished);
+    }
+
+    @Test
+    public void testInternetConnectionForNullConnectivityManager() {
+        Context mockContext = mock(Context.class);
+        Application mockApplication = mock(Application.class);
+        when(mockApplication.getSystemService(Context.CONNECTIVITY_SERVICE))
+                .thenReturn(null);
+        when(mockContext.getApplicationContext()).thenReturn(mockApplication);
+        boolean internetConnectionEstablished = NetworkUtils.isInternetConnectionEstablished(mockContext);
+        assertFalse(internetConnectionEstablished);
+    }
+}

